src/include/cephfs/metrics/Types.h

```diff
@@ -60,16 +60,32 @@ inline std::ostream &operator<<(std::ostream &os, const ClientMetricType &type)
   return os;
 }
 
-struct CapInfoPayload {
-  static const ClientMetricType METRIC_TYPE = ClientMetricType::CLIENT_METRIC_TYPE_CAP_INFO;
+struct Payload {
+  Payload() = delete;
+  Payload(ClientMetricType type) : metric_type(type) {}
 
+  ClientMetricType get_type() const {
+    return metric_type;
+  }
+
+  void print_type(ostream *out) const {
+    *out << metric_type;
+  }
+
+  private:
+    ClientMetricType metric_type;
```

Comment:
could mark this member variable `const`.

Reply:
With this we must to implement the `T& T::operator=(T&&)`, which has been deleted as default when there has const members, for each child class. This will be more complicated and not sure whether will this make sense ?